packages/compiler-cli/src/ngtsc/typecheck/extended/src/template_checks/banana_in_box/index.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {TmplAstBoundEvent, TmplAstNode, TmplAstRecursiveVisitor} from '@angular/compiler';
+import * as ts from 'typescript';
+import {ErrorCode} from '../../../../../diagnostics';
+import {TemplateCheck, TemplateContext, TemplateDiagnostic} from '../../../api/api';
+
+/**
+ * Ensures the two-way binding syntax is correct.
+ * Parentheses should be inside the brackets "[()]".
+ * Will return diagnostic information when "([])" is found.
+ */
+export class BananaInBoxCheck implements TemplateCheck<ErrorCode.BANANA_IN_BOX> {
+  code: ErrorCode.BANANA_IN_BOX = 8101;
+
+  run(ctx: TemplateContext,
+      template: TmplAstNode[]): TemplateDiagnostic<ErrorCode.BANANA_IN_BOX>[] {
+    const visitor = new BananaVisitor(ctx);
+
+    return visitor.getDiagnostics(template);
+  }
+}
+
+class BananaVisitor extends TmplAstRecursiveVisitor {
+  private diagnostics: ts.Diagnostic[] = [];
+  readonly ctx: TemplateContext;
+
+  constructor(ctx: TemplateContext) {
+    super();
+    this.ctx = ctx;
+  }
```

Comment:
We can probably do something like this:
```suggestion
  constructor(readonly public ctx: TemplateContext) {
    super();
  }
```